subsys/net/lib/wifi_mgmt_ext/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+zephyr_library()
+zephyr_library_sources(wifi_mgmt_ext.c)
+
+zephyr_library_include_directories(${NCS_DIR}/modules/lib/hostap/wpa_supplicant/)
+zephyr_library_include_directories(${NCS_DIR}/modules/lib/hostap/src/)
+zephyr_library_include_directories(${NCS_DIR}/modules/lib/hostap/src/utils)
+
+zephyr_include_directories(
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/zephyr/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/osal/utils/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/osal/os_if/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/osal/bus_if/bus/qspi/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/osal/bus_if/bal/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/osal/fw_if/umac_if/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/osal/fw_load/mips/fw/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/osal/hw_if/hal/inc
+	${NCS_DIR}/main/nrf/drivers/wifi/nrf700x/zephyr/src/qspi/inc
+)
```

Comment:
This I totally don't understand, why would this lib extend global include directory list with those driver internals?

Reply:
Does this extend the global include directory list? The wifi driver has the same thing: https://github.com/nrfconnect/sdk-nrf/blob/8c784a4979b8a46e88ce72dad1a7dc12a2649c5a/drivers/wifi/nrf700x/CMakeLists.txt#L9
We need these internal include dirs because some wpa_supplicant headers depend on those. I can include them with zephyr_library_include_directories instead.

Reply:
`zephyr_include_directories()` extends the system-wide include list. Anyway, adding any directories from supplicant/driver to the include list here is a hacky workaround, all public headers should be exposed by the driver/supplicant themself.

Another thing is that I don't think the path is correct anyway, why is there `main` in the path? I don't have it in my local installment. It seems, that if we decide to keep this workaround, it'd be better to use `{NRF_DIR}` instead, which leads directly to `sdk-nrf` directory.

Reply:
The thing is that I have to use a private header of `wpa_supplicant`. Even if I used the `wpa_supplicant_ctrl_iface` functions, I still need `wpa_supplicant_i.h`, which in turn needs several other headers. But It's true that I don't have to add those directories here since the driver already adds all of them globally (which is probably unintended).

Reply:
this is a TODO that needs to be resolved before we can merge this. will have to wait for this dependency: https://github.com/nrfconnect/sdk-hostap/pull/32